cheri-c-programming.tex

```diff
@@ -896,7 +896,21 @@ \subsubsection{Out-of-bounds pointers}
 
 \note{I feel like this section wants a reference to CHERI Concentrate?}{nwf}
 
-The capability compression model exploits redundancy between the pointer and
+ISO C permits pointers to go only one byte beyond their original
+allocation, but real code often \psnote{not sure how strong to be
+  there} constructs transient pointer values which are further out of
+bounds.  To support this, capabilities can hold a range of
+out-of-bounds addresses while retaining a valid tag.  Dereferencing an
+out-of-bounds pointer will raise a hardware exception (see
+\Cref{sec:faults}).  However, an out-of-bounds pointer can be
+dereferenced once it has been brought back in bounds by adjusting the
+address or suppling a suitable offset in the dereference.  This means
+that operations which modify the address of a capability do not clear
+the tag for out-of-bounds addresses.  Instead, a failure occurs if an
```

Comment:
Maybe slightly easier to understand:
```suggestion
that operations which modify the address of a capability do not clear
the tag if the resulting address is out-of-bounds.  Instead, a failure occurs if an
```

Reply:
I do think this is clearer, but I've axed the sentence when adopting your earlier change.  If we bring it back we can add this fix to it.